src/OrleansRuntime/GrainDirectory/GrainDirectoryHandoffManager.cs

```diff
@@ -38,10 +38,10 @@ internal List<ActivationAddress> GetHandedOffInfo(GrainId grain)
             foreach (var partition in directoryPartitionsMap.Values)
             {
                 var result = partition.LookUpGrain(grain);
-                if (result != null)
+                if (result.Addresses != null)
                 {
                     // Force the list to be created in order to avoid race conditions
-                    return result.Item1.Select(pair => ActivationAddress.GetAddress(pair.Item1, grain, pair.Item2)).ToList();
+                    return result.Addresses.Select(a => ActivationAddress.GetAddress(a.Silo, grain, a.Activation, a.Status)).ToList();
```

Comment:
Any reason to recreate the list? It doesn't seem to be accomplishing anything. It creates an identical `List<ActivationAddress>` as `result.ActivationAddresses`